libraries/botbuilder-ai/src/luisRecognizer.ts

```diff
@@ -417,40 +428,84 @@ export class LuisRecognizer implements LuisRecognizerTelemetryClient {
         telemetryProperties?: Record<string, string>,
         telemetryMetrics?: Record<string, number>,
         options?: LuisRecognizerOptionsV2 | LuisRecognizerOptionsV3 | LuisPredictionOptions
+    ): Promise<RecognizerResult>;
+
+    /**
+     * Calls LUIS to recognize intents and entities in a users utterance.
+     *
+     * @param {string} utterance The utterance to be recognized.
+     * @param {LuisRecognizerOptionsV2 | LuisRecognizerOptionsV3 | LuisPredictionOptions} options (Optional) options object used to override control predictions. Should conform to the [LuisRecognizerOptionsV2] or [LuisRecognizerOptionsV3] definition.
+     */
+    public async recognize(
+        utterance: string,
+        options?: LuisRecognizerOptionsV2 | LuisRecognizerOptionsV3 | LuisPredictionOptions
+    ): Promise<RecognizerResult>;
+
+    /**
+     * @internal
+     */
+    public async recognize(
+        contextOrUtterance: DialogContext | TurnContext | string,
+        maybeTelemetryPropertiesOrOptions?:
+            | Record<string, string>
+            | LuisRecognizerOptionsV2
+            | LuisRecognizerOptionsV3
+            | LuisPredictionOptions,
+        maybeTelemetryMetrics?: Record<string, number>,
+        maybeOptions?: LuisRecognizerOptionsV2 | LuisRecognizerOptionsV3 | LuisPredictionOptions
     ): Promise<RecognizerResult> {
-        const turnContext = context instanceof DialogContext ? context.context : context;
-        const cached = turnContext.turnState.get(this.cacheKey);
-        const luisRecognizer = options ? this.buildRecognizer(options) : this.luisRecognizerInternal;
-        if (!cached) {
-            const utterance = turnContext.activity.text || '';
-            let recognizerPromise: Promise<RecognizerResult>;
-
-            if (!utterance.trim()) {
-                // Bypass LUIS if the activity's text is null or whitespace
-                recognizerPromise = Promise.resolve({
-                    text: utterance,
-                    intents: { '': { score: 1 } },
-                    entities: {},
-                });
-            } else {
-                recognizerPromise = luisRecognizer.recognizeInternal(context);
-            }
+        // This type check, when true, logically implies that the function is being invoked as the two-argument string + optional options overload variant.
+        if (typeof contextOrUtterance === 'string') {
+            const utterance = contextOrUtterance;
+            const options = UnsafeLuisRecognizerUnion.optional().check(maybeTelemetryPropertiesOrOptions);
+
+            const luisRecognizer = options ? this.buildRecognizer(options) : this.luisRecognizerInternal;
+
+            return luisRecognizer.recognizeInternal(utterance);
+        } else {
+            const telemetryProperties = t
+                .Dictionary(t.String, t.String)
+                .optional()
+                .nullable()
```

Comment:
Is `nullable` also required? I figured `optional` would be sufficient.

Reply:
18 different tests break without it with `ValidationError: Expected { [_: string]: string }, but was null`. All of the failing tests are [the JSON kind](https://github.com/microsoft/botbuilder-js/tree/main/libraries/botbuilder-ai/tests/TestData/LuisRecognizerV3) so I thought it best to add the `nullable()` in case changing the test JSON and then not having it might break something I don't understand.

Per the `runtypes` docs, it looks like `optional()` makes the properties of a Record optional, but not the record, itself.

Reply:
Calling `optional()` on a `runtype` essentially appends `| undefined` to the type. 

![Screen Shot 2021-05-17 at 12 28 23 PM](https://user-images.githubusercontent.com/1235378/118545395-61e73e00-b70b-11eb-8592-9364975486ac.png)

It looks like calling `nullable()` essentially appends `| null`.

![Screen Shot 2021-05-17 at 12 29 22 PM](https://user-images.githubusercontent.com/1235378/118545482-84795700-b70b-11eb-9ed3-640a07105c15.png)

Regardless, it does seem like `nullable()` is important! Good catch.